src/js/DropContainer.jsx

```diff
@@ -0,0 +1,281 @@
+      * @param  {Event]} event
+   */
+   onDragOver (e) {
+      e.preventDefault();
+      e.stopPropagation();
+      e.dataTransfer.dropEffect = 'copy';
+
+      return false;
+   }
+
+   /**
+      * Handle the drag leave event
+   */
+   onDragLeave (e) {
+      e.preventDefault();
+
+      // Only deactivate once the dropzone and all children was left.
+      if (--this.enterCounter > 0) {
+         return;
+      }
+
+      this.setState({
+         isDragActive: false,
+         isDragReject: false,
+         borderStyle: '',
+         visibility: 'show'
+      });
+
+      if (this.props.onDragLeave) {
+         this.props.onDragLeave.call(this, e);
+      }
+
+      if(this.state.imageFile){
+         // console.log('onDragLeave');
+         return;
+      }
+  }
+
+   /**
+      * Handle the drop event
+      * @param  {Event]} event
+   */
+   onDrop (e) {
+      e.preventDefault();
+      this.setState({
+         isDragActive: false,
+         borderStyle: '',
+         visibility: 'show'
+      });
+      this.handleChange.call(this, e);
+   }
+
+   /**
+      * Handle the click event on drop container
+   */
+   // onClick () {
+   //    console.log('on click called');
+   //    this.refs.fileInput.myTextInput.click();
+   // }
+
+   /**
+      * Handle the image state
+      * If image is available, will create a URL obj
+      * And modify the state value
+      * @param  {Event]} event
+   */
+   handleChange (event) {
+      let e = event.nativeEvent, imageFile;
+
+      const dataTransferItems = this.extractData(event);
+      const allFilesAccepted = this.fileAccepted(Array.prototype.slice.call(dataTransferItems));
+
+      // Check to make sure the dropped file
+      // is in image format - image/*
+      if(allFilesAccepted){
+         // Setting the state image source
+         // With drag & drop event
+         if(e.dataTransfer) {
+            let evt = e.dataTransfer.files[0];
+
+            if(evt !== undefined) {
+               imageFile = URL.createObjectURL(evt);
+               // console.log('**DRAGGED** SELECTED FILE', evt);
+               this.setState({
+                  imageFile: imageFile
+               });
+            } else {
+               console.warn('Warning: no image selected');
+            }
+
+         // Setting the state image source
+         // With click button event
+         } else {
+            let evt = event.target.files[0];
+
+            // Image src to pass it to ImagePreview Component
+            if(evt !== undefined) {
+               imageFile = URL.createObjectURL(event.target.files[0]);
+               // console.log('SELECTED FILE', event.target.files[0]);
+
+               this.setState({
+                  imageFile: imageFile
+               });
+            } else {
+               console.warn('Warning: no image selected');
+            }
+         }
+      }
+   }
+
+   fileAccepted(file) {
+      return file.every(f => accepts(f, 'image/*'));
```

Comment:
'image/*' shouldn't be hard coded. We may want to upload other file types like pdfs or documents